django_guid/middleware.py

```diff
@@ -146,3 +144,13 @@ def _get_id_from_header(self, request: HttpRequest) -> str:
             )
 
         return request.correlation_id
+
+
+@receiver(request_finished)
+def clean_finished_request_guid(sender: Optional[dict], **kwargs) -> None:
```

Comment:
If the `_delete_guid` log is called, the signal would have to be fired, because it is no longer called from anywhere else, so think that will be sufficient for testing.

Signals.py seems to cause import errors, because it doesn't fire unless it's imported - maybe there's another way to structure it that makes sense.

Reply:
Added a debug logger to the signals functions, so now we're testing that the signal is fired, and that the GUID is deleted.